File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque
n,m,v = map(int,input().split())
graph = [[]for i in range(n+1)]
for i in range(m): #간선의 개수
    a,b = map(int,input().split())
    graph[a].append(b)
    graph[b].append(a)
    graph[a].sort()
    graph[b].sort()

#graph = [[],[2,3,4],[1,4],[1],[1,2]]

def dfs(graph,v,visited):
    visited[v] = True  #방문 node 방문 처리
    print(v,end=' ')
    for j in graph[v]: #graph v 에 있는 node들을 하나씩 방문
        if not visited[j]: # 예를들어 [2,3,4] 중 2가 visited[2] 가 False 라면 => dfs(graph,2,visited) 로 다시 돌리기
            dfs(graph,j,visited)

def bfs(graph,start,visited):
    queue = deque([start])

    visited[start] = True

    while queue:
        v = queue.popleft()
        print(v, end=' ')
        for k in graph[v]:
            if not visited[k]:
                queue.append(k)
                visited[k] = True
visited = [False] * (n+1)
dfs(graph, v, visited)
print("")
visited = [False] * (n+1)
bfs(graph, v, visited)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.js
let readLine = require("readline");

let rl = readLine.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let N, M, V;
let arr;
let inputLine = 0;
let visited1;
let visited2;
rl.on("line", function (line) {
  if (inputLine == 0) {
    let cmd = line.split(" ").map((value) => parseInt(value));
    N = cmd[0];
    M = cmd[1];
    V = cmd[2];
    arr = Array(N + 1)
      .fill()
      .map(() => Array());
    visited1 = Array(N + 1).fill(false);
    visited2 = Array(N + 1).fill(false);
  } else if (inputLine >= 1 && inputLine <= M) {
    let cmd = line.split(" ").map((value) => parseInt(value));
    arr[cmd[0]].push(cmd[1]);
    arr[cmd[1]].push(cmd[0]);
  }
  if (inputLine == M) {
    let answerBfs = [];
    let bfs = (v) => {
      visited1[v] = true;
      let Queue = [];
      Queue.push(v);
      while (!Queue.length == 0) {
        let node = Queue.shift();
        answerBfs.push(node);
        arr[node].sort((a, b) => a - b);
        for (let i = 0; i < arr[node].length; i++) {
          if (!visited1[arr[node][i]]) {
            Queue.push(arr[node][i]);
            visited1[arr[node][i]] = true;
          }
        }
      }
    };
    let answerDfs = [];
    let dfs = (v) => {
      visited2[v] = true;
      answerDfs.push(v);
      arr[v].sort((a, b) => a - b);
      for (let i = 0; i < arr[v].length; i++) {
        if (!visited2[arr[v][i]]) {
          dfs(arr[v][i]);
        }
      }
    };
    bfs(V);
    dfs(V);
    console.log(answerDfs.join(" "));
    console.log(answerBfs.join(" "));
    process.exit();
  }
  inputLine++;
});
